File: app/src/main/java/com/example/movieslisttask/viewmodel/MoviesViewModel.java
package com.example.movieslisttask.viewmodel;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.example.movieslisttask.model.Movie;
import com.example.movieslisttask.model.MovieDBRepository;
import com.example.movieslisttask.model.MovieRepository;

import java.util.List;

/**
 * Created by alexkorolov on 2020-01-17.
 */
public class MoviesViewModel extends AndroidViewModel {

    private MovieDBRepository dbRepository;
    private MovieRepository movieRepository;

    public MoviesViewModel(Application application) {
        super(application);
        dbRepository = new MovieDBRepository(application);
        movieRepository = new MovieRepository(application);
    }

    public LiveData<List<Movie>> getallMovies() {
        movieRepository.getMoviesFromAPI();
        return dbRepository.getAllMovies();
    }

    public LiveData<List<Movie>> getAllFavorites() {
        return dbRepository.getAllFavoriteMovies();
    }
}

File: app/src/main/java/com/example/movieslisttask/ui/movies_list/MoviesViewModel.kt
package com.example.movieslisttask.ui.movies_list

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.movieslisttask.db.entities.Movie
import com.example.movieslisttask.model.MovieRepository
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

/**
 * Created by alexkorolov on 2020-01-17.
 */
class MoviesViewModel(private val movieRepository: MovieRepository) : ViewModel() {

    fun getMovies(isFavorite: Boolean): LiveData<List<Movie>> {
        return if (isFavorite) {
            movieRepository.getFavoriteMovies()
        } else {
            viewModelScope.launch(Dispatchers.IO) {
                movieRepository.fetchPopularMoviesFromAPI()
            }
            movieRepository.getPopularMovies()
        }
    }

    fun handleFavoriteMovie(movie: Movie) {
        viewModelScope.launch(Dispatchers.IO) {
            movieRepository.updateMovie(movie)
        }
    }
}